crates/release_plz_core/src/published_packages/git_tags.rs

```diff
@@ -0,0 +1,368 @@
+                tracing::debug!("skipping {CARGO_LOCK} from repo in favor of the one we generate");
+                continue;
+            }
+
+            let dest_path = target_dir.as_std_path().join(relative_path);
+
+            tracing::trace!(
+                "copying package file {} to {}",
+                package_file.display(),
+                dest_path.display()
+            );
+
+            fs_err::create_dir_all(dest_path.parent().context("bug: dest_path has no parent")?)?;
+            fs_err::copy(&package_file, &dest_path)?;
+
+            published_package_files.push(
+                relative_path
+                    .to_path_buf()
+                    .try_into()
+                    .context("package contains non-UTF-8 path")?,
+            );
+
+            // We compare the local package's Cargo.toml with the original Cargo.toml
+            // (stored in Cargo.toml.orig) of the published package.
+            // Copy the Cargo.toml to Cargo.toml.orig
+            if relative_path_os_str == cargo_utils::CARGO_TOML {
+                tracing::trace!("Copying Cargo.toml to Cargo.toml.orig");
+
+                fs_err::copy(&package_file, target_dir.join(CARGO_TOML_ORIG))?;
+                published_package_files.push(CARGO_TOML_ORIG.into());
+            }
+        }
+
+        if package.include_lockfile() {
+            // Generate the lock file to be included in the package
+            // See cargo::ops::cargo_package::build_lock
+
+            // Use resolve information already contained in the workspace lock file.
+            let orig_resolve = cargo::ops::load_pkg_lockfile(&workspace)?;
+
+            // Create an ephemeral workspace containing the single package we are packaging
+            // This prevents packages in the workspace lock file that not required by this package
+            // from being included in the generated lock file.
+            let resolve_ws = Workspace::ephemeral(package.clone(), &gctx, None, true)?;
+            let mut package_registry = resolve_ws.package_registry()?;
+
+            let new_resolve = cargo::ops::resolve_with_previous(
+                &mut package_registry,
+                &resolve_ws,
+                &cargo::core::resolver::CliFeatures::new_all(true),
+                cargo::core::resolver::features::HasDevUnits::Yes,
+                orig_resolve.as_ref(),
+                None,
+                &[],
+                true,
+            )?;
+
+            let lockfile_contents = cargo::ops::resolve_to_string(&resolve_ws, &new_resolve)?;
+            let cargo_lock_file_name = Utf8Path::new(CARGO_LOCK).to_path_buf();
+            fs_err::write(target_dir.join(&cargo_lock_file_name), &lockfile_contents)?;
+
+            published_package_files.push(cargo_lock_file_name);
+            tracing::trace!("wrote {CARGO_LOCK} contents:\n{}", lockfile_contents);
+        }
+
+        Ok(published_package_files)
+    }
```

Comment:
This part of the code seems to be missing some logic for resolving workspace references. When trying the current version of the code, the `release-pr` subcommand fails when the publish package has workspace references.

None of the workspace references in `[package]` or `[dependency]` seem to resolve. There is also no difference between the `Cargo.toml` and the `Cargo.toml.orig` in the generated tmp folder. This leads to a failure in <https://github.com/release-plz/release-plz/blob/af0b46b6babcb69b4ae6857a01227dd4b4868d50/crates/release_plz_core/src/published_packages/git_tags.rs#L215-L220> because cargo cannot read the workspace root.

Reply:
Ah yeah, I chose to skip the normalization Cargo performs because it also removes path dependencies, but missed this possibility. I'll look into adding the manifest normalization without touching path dependencies, and also add workspace references to a test so this behavior is tested for. Thanks for pointing that out!

Reply:
This turned out to be easier than I thought - the path dependency stripping happens in a separate step right before packaging, so I was able to just used the normalized manifest generated by `cargo` when loading a manifest. This should be fixed according to the tests I added, but can you please run your workflow again to ensure that it works?

Reply:
With the new changes (`843c55c2781bf460a0f1d6e10de08bf57bdbc19c`) my workflow works (for reference, I do not publish to a registry (all cargo manifests have `package.publish = false`), and have one package that I do not want git tags for.

The relevant excerpts from my release-plz config look like

```toml
[workspace]
publish = false
release = true

[[package]]
name = "workspace-hack"
release = false
```

I have not tested without setting `workspace.publish` to `false`, let me know if you want me to

Reply:
Thanks! If you can test with `workspace.publish` set to false, that would also be great. It should work without any issues, but would still be good to confirm.

Reply:
Without setting `[workspace.publish]` in the `release-plz.toml`, the `package.publish` from the `Cargo.toml` seems to be respected 👍 (though I can only test the `package.publish = false` case, since I don't have  any packages that publish to a registry)